Feature/tf df serving

picousse opened the pull request:
Basic tip on the TF-DF TF Serving docker container

julienschuermans commented on structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:
```diff
@@ -0,0 +1,50 @@
+# Serving TF-DF models using TF Serving
+
+*This is a small excerpt from the  https://blog.ml6.eu/serving-decision-forests-with-tensorflow-b447ea4fc81c blogpost*
```
```suggestion
*This is a small excerpt from our [blogpost on decision forests](https://blog.ml6.eu/serving-decision-forests-with-tensorflow-b447ea4fc81c)*
```

julienschuermans commented on structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:
```diff
@@ -0,0 +1,50 @@
+
+TensorFlow and TensorFlow Extended are a broad ecosystem of software packages, allowing to train, serve,... ML models.
+
+The addition of decision forests to TensorFlow (blogpost)[https://www.tensorflow.org/decision_forests], increases the applicability of TensorFlow on many more use cases.
```
```suggestion
The [addition of decision forests to TensorFlow](https://www.tensorflow.org/decision_forests) increases the applicability of TensorFlow on many more use cases.
```

julienschuermans commented on structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:
```diff
@@ -0,0 +1,50 @@
+Training is only one part of ML models: robust inference calculations are as important in a data pipeline.
+This is where TensorFlow Serving comes in. TensorFlow Serving basically creates an API around a model. On top of that it allows to split code and model, enabling separate versioning.
+
+Unfortunately, when TensorFlow Decision Forests (TF-DF) was released, TensorFlow Serving was not ready to except these new models.
```
```suggestion
Unfortunately, when TensorFlow Decision Forests (TF-DF) was released, TensorFlow Serving was not ready to serve these new models.
```

julienschuermans commented on structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:
```diff
@@ -0,0 +1,50 @@
+
+Unfortunately, when TensorFlow Decision Forests (TF-DF) was released, TensorFlow Serving was not ready to except these new models.
+
+As we found TF-DF models to be very interesting, we, at ML6, decided to build a custom docker image to serve these models.
```
```suggestion
We decided to build a custom `tf-serving` docker image to serve these models.
```

julienschuermans commented on structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:
```diff
@@ -0,0 +1,50 @@
+
+As we found TF-DF models to be very interesting, we, at ML6, decided to build a custom docker image to serve these models.
+
+The docker image can be found at (https://hub.docker.com/r/ml6team/tf-serving-tfdf)[https://hub.docker.com/r/ml6team/tf-serving-tfdf].
```
```suggestion
The docker image can be found on [DockerHub](https://hub.docker.com/r/ml6team/tf-serving-tfdf).
```

julienschuermans commented on structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:
```diff
@@ -0,0 +1,50 @@
+
+The docker image can be found at (https://hub.docker.com/r/ml6team/tf-serving-tfdf)[https://hub.docker.com/r/ml6team/tf-serving-tfdf].
+
+To use the docker image for a penguin model, the following code can be used:
```
Maybe refer to the colab that was used to train this penguin model?

picousse replied:
has been added

julienschuermans reviewed (commented):
Nice work, I've fixed some hyperlinks and added a simple comment.
Can you also add a line to the `quick_tips/structured_data/README.md` with a reference to this new `2021_07_02_TFDF_serving` directory?

Sorry for the late response!

picousse commented:
@julienschuermans I think I applied all your changes, added the colab link and added the link.
Ready to be merged if you ask me.

Do you want to check before merging?

julienschuermans commented on structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:
```diff
@@ -0,0 +1,51 @@
+We decided to build a custom `tf-serving` docker image to serve these models.
+
+The docker image can be found on [DockerHub](https://hub.docker.com/r/ml6team/tf-serving-tfdf).
+The model used in the following example was trained using the [TF-DF pinguins colab](https://colab.research.google.com/github/tensorflow/decision-forests/blob/main/documentation/tutorials/beginner_colab.ipynb).
```
```suggestion
The model used in the following example was trained using the [TF-DF penguins colab](https://colab.research.google.com/github/tensorflow/decision-forests/blob/main/documentation/tutorials/beginner_colab.ipynb).
```

julienschuermans commented on structured_data/README.md:
```diff
@@ -12,4 +12,7 @@ This notebook provides looks into how MINIROCKET — a novel feature extraction
-to track anomalous changes in your data.
\ No newline at end of file
+to track anomalous changes in your data.
+- [_TF Decision Forest: Serving TF-DF models with TF Serving_ (02/07/2021)](2021_07_02_TFDF_serving):
```
```suggestion
- [_Tabular Data: Serving decision forests with Tensorflow Serving_ (02/07/2021)](2021_07_02_TFDF_serving):
```

julienschuermans commented on structured_data/README.md:
```diff
@@ -12,4 +12,7 @@ This notebook provides looks into how MINIROCKET — a novel feature extraction
\ No newline at end of file
+to track anomalous changes in your data.
+- [_TF Decision Forest: Serving TF-DF models with TF Serving_ (02/07/2021)](2021_07_02_TFDF_serving):
+  This Readme documents the use of the custom build TF Serving docker build to serve TF-Decision Forest
```
```suggestion
  This Readme documents the use of our custom TF Serving docker image built to serve Tensorflow Decision Forests.
```

julienschuermans commented on structured_data/README.md:
```diff
@@ -12,4 +12,7 @@ This notebook provides looks into how MINIROCKET — a novel feature extraction
+to track anomalous changes in your data.
+- [_TF Decision Forest: Serving TF-DF models with TF Serving_ (02/07/2021)](2021_07_02_TFDF_serving):
+  This Readme documents the use of the custom build TF Serving docker build to serve TF-Decision Forest
+  models.
```
```suggestion
```

julienschuermans commented:
LGTM, let's merge it!
